zero/src/provider.rs

```diff
@@ -1,15 +1,29 @@
+use std::future::Future;
 use std::ops::{Deref, DerefMut};
 use std::sync::Arc;
 
 use alloy::primitives::BlockHash;
 use alloy::rpc::types::{Block, BlockId, BlockTransactionsKind};
 use alloy::{providers::Provider, transports::Transport};
 use anyhow::Context;
+use mockall::automock;
 use tokio::sync::{Mutex, Semaphore, SemaphorePermit};
 
+use crate::rpc::RpcType;
+
 const CACHE_SIZE: usize = 1024;
 const MAX_NUMBER_OF_PARALLEL_REQUESTS: usize = 128;
 
+#[automock]
+pub trait BlockProvider {
```

Comment:
I would prefer to name it `ZeroBlockProvider`, not to be mixed with the providers defined in `alloy` libraries.

Reply:
Updated